The SSB pcicore driver does create some MMIO resource collisions.
However, the pcicore PCI-fixup routine fixes these collisions afterwards.
Remove this sanity check for now until we find a better solution.
--mb
--- a/arch/mips/pci/pci.c
+++ b/arch/mips/pci/pci.c
@@ -177,10 +177,8 @@
 			continue;
 
 		r = &dev->resource[idx];
-		if (!r->start && r->end) {
-			printk(KERN_ERR "PCI: Device %s not available because of resource collisions\n", pci_name(dev));
-			return -EINVAL;
-		}
+		if (!r->start && r->end)
+			printk(KERN_WARNING "PCI: Device %s resource collisions detected. Ignoring...\n", pci_name(dev));
 		if (r->flags & IORESOURCE_IO)
 			cmd |= PCI_COMMAND_IO;
 		if (r->flags & IORESOURCE_MEM)
